### PR TITLE
EVM module:  add `eth_getTransactionByHash` endpoint

### PR DESCRIPTION
**Pull request by bkolad**
# Description
This PR includes the following changes:
1. Adds a dependency on `anvil-core` to enable reuse of core Ethereum types in EVM module endpoints.
2. Improves the implementation of `eth_sendRawTransaction` by utilizing `anvil-core`.
3. Adds an `eth_getTransactionByHash` endpoint (currently the implementation is very naive, we save all submitted transactions in the EVM module so we can fetch them with  `eth_getTransactionByHash`).

The todos in the code will be fixed in following PRs.

## Linked Issues
- Related to #413

## Testing
Tested manually with `demo-rollup`

### CODE REVIEW

File: examples/demo-rollup/rollup_config.toml [hunk 1]
```diff
@@ -4,7 +4,7 @@ start_height = 1
 
 [da]
 # The JWT used to authenticate with the celestia light client. Instructions for generating this token can be found in the README
-celestia_rpc_auth_token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJwdWJsaWMiLCJyZWFkIiwid3JpdGUiLCJhZG1pbiJdfQ.yEqKNCDpXwddRTQYQeMNkQ59O22LI95CS-HeZMcoN-k"
+celestia_rpc_auth_token = "eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJBbGxvdyI6WyJwdWJsaWMiLCJyZWFkIiwid3JpdGUiLCJhZG1pbiJdfQ.jp_hghhRkcur3g0o-QhQnkLt89YPu4oykLtZW52_ZA0"
```
Comment (citizen-stig) [inline examples/demo-rollup/rollup_config.toml:7]: @dubbelosix  =)

File: examples/demo-stf/src/sov-cli/test_data/tx_blob
Comment (citizen-stig) [inline examples/demo-stf/src/sov-cli/test_data/tx_blob:1]: Can we remove it?

File: module-system/module-implementations/sov-evm/src/lib.rs [hunk 2]
```diff
@@ -60,6 +60,9 @@ mod experimental {
 
         #[state]
         pub(crate) block_env: sov_state::StateValue<BlockEnv>,
+
+        #[state]
+        pub(crate) transactions: sov_state::StateMap<[u8; 32], EvmTransaction>,
```
Comment (citizen-stig) [inline module-system/module-implementations/sov-evm/src/lib.rs:65]: Shoult it be `Bytes32` type alias or is it different?

File: Cargo.toml [hunk 3]
```diff
@@ -73,3 +73,9 @@ toml = "0.7.3"
 jsonrpsee = "0.16.2"
 tempfile = "3.5"
 tokio = { version = "1", features = ["full"] }
+
+ethers = { git = "https://github.com/gakonst/ethers-rs", default-features = false }
+anvil  = { git = "https://github.com/foundry-rs/foundry", version = "0.1.0" }
+#anvil = { path = "../foundry/anvil" }
```
[the remark this answers is not in the document]
Comment (bkolad) [reply Cargo.toml]: done

File: examples/demo-stf/src/sov-cli/test_data/tx_blob
Comment (bkolad) [reply examples/demo-stf/src/sov-cli/test_data/tx_blob:1]: done

**Pull request merged**